crates/ruff_python_ast/src/nodes.rs

```diff
@@ -2578,13 +2577,235 @@ impl Ranged for Identifier {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub enum Number {
+    /// A "small" number that can be represented as an `i64`.
+    Small(i64),
+    /// A "large" number that cannot be represented as an `i64`.
+    Big(Box<str>),
+}
+
+impl std::fmt::Display for Number {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Number::Small(value) => write!(f, "{value}"),
+            Number::Big(value) => write!(f, "{value}"),
+        }
+    }
+}
+
+/// A Python integer literal. Represents both small (fits in an `i64`) and large integers.
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct Int(Number);
+
+impl Int {
+    pub const ZERO: Int = Int(Number::Small(0));
+    pub const ONE: Int = Int(Number::Small(1));
```

Comment:
These are sort of random but they allow you to pattern match against `0` and `1`, which end up being the only values that we pattern-match against.